test/tools/d_do_test.d

```diff
@@ -816,13 +816,20 @@ int tryMain(string[] args)
             if (envData.autoUpdate)
             if (auto ce = cast(CompareException) e)
             {
-                auto existingText = input_file.readText;
-                auto updatedText = existingText.replace(ce.expected, ce.actual);
-                std.file.write(input_file, updatedText);
                 // remove the output file in test_results as its outdated
-                f.close();
```

Comment:
N.B.: It's not necessary to close the file manually - it's ref-counted and will be closed as soon as the scope is left.